forge-gui/res/cardsfolder/upcoming/vigorous_farming.txt

```diff
@@ -0,0 +1,10 @@
+Name:Vigorous Farming
+ManaCost:2 G
+Types:Enchantment
+R:Event$ Moved | ValidCard$ Land.YouCtrl | Destination$ Battlefield | ReplaceWith$ ETBUntapped | ReplacementResult$ Updated | ActiveZones$ Battlefield | Description$ Lands you control enter the battlefield untapped.
+SVar:ETBUntapped:DB$ Untap | ETB$ True | Defined$ ReplacedCard
+T:Mode$ Phase | Phase$ End of Turn | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigAnimate | TriggerDescription$ At the beginning of your end step, the topmost land card in your library perpetually gains "Whenever this land is tapped for mana, add an additional {G}."
+SVar:TrigAnimate:DB$ Animate | Defined$ Land.TopLibraryLand+YouCtrl | Triggers$ ManaTap | Duration$ Perpetual
```

Comment:
it probably doesn't work because your Defined syntax is incomplete

Reply:
Sure enough, added `ValidLibrary` to that argument and now it works 👍